app/routes.js

```diff
@@ -94,13 +93,21 @@ export const requireAuthAndNoPatient = (api, store) => (nextState, replace, cb)
 };
 
 /**
+ * This function ensures any logged in state is destroyed on entering a route
+ * 
+ * @param  {Object} nextState
+ * @param  {Function} replace
+ */
+export const ensureNoAuth = (api) => (nextState, replace) => {
+  api.user.logout();
```

Comment:
So we're relying on the fact that `api.user.logout` is synchronous here (but does take an optional cb). I believe elsewhere in this file we're passing the cb...shall we align one way or the other, just for reducing confusion?